### PR TITLE
Subscriptions refactor

### PR DESCRIPTION
**Pull request by drusepth**
Work in progress, not ready to merge and needs tested

### CODE REVIEW

File: app/services/subscription_service.rb [hunk 1]
```diff
@@ -0,0 +1,145 @@
+class SubscriptionService < Service
```
Comment (mpigsley) [inline app/services/subscription_service.rb]: Controllers to Services! Love it!

File: app/services/subscription_service.rb [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  STARTER_PLAN_IDS = [1]
+  FREE_FOR_LIFE_PLAN_IDS = [2]
+  PAID_PREMIUM_PLAN_IDS = [3, 4, 5, 6]
+
+  SUBSCRIPTION_LENGTH = 5.years #todo refactor this out
+
+  def self.transition_plan user:, old_plan:, new_plan:, stripe_object: nil
+    # Fetch the user's current Stripe information
+    stripe_object ||= SubscriptionService.stripe_customer_object_for(current_user)
```
Comment (mpigsley) [inline app/services/subscription_service.rb]: Any chance this could still return nil?

File: app/controllers/subscriptions_controller.rb [hunk 3]
```diff
@@ -264,7 +138,7 @@ def information_change
         end_date: Time.now.end_of_day + 31.days
       )
 
-      report_subscription_change_to_slack current_user, old_billing_plan, new_billing_plan
+      SubscriptionService.report_subscription_change_to_slack current_user, old_billing_plan, new_billing_plan
```
Comment (mpigsley) [inline app/controllers/subscriptions_controller.rb]: Ooo didn't realize this was a thing!

**Comment by drusepth** [comment]
Replaced by #263

**Pull request closed without merging**